client/state/data-layer/wpcom-http/actions.js

```diff
@@ -64,6 +74,8 @@ export const http = (
 		onSuccess: onSuccess || action,
 		onFailure: onFailure || action,
 		onProgress: onProgress || action,
+		processResponseInStreamMode,
```

Comment:
why do we need a boolean _and_ a function? if the function is present, why would need to set the flag?

will `processResponseInStreamMode !== ('function' === typeof onStreamRecord)` ever?

Reply:
good point! not in this patch, so i’ve dropped the flag from this layer in 1e53809ca256bdca56139072f9d9a0c6a4880b5b.

but #47614 has to undo this, because we add on­Stream­Record to dispatch­Request. for requests that use dispatch­Request, we point on­Stream­Record to the default action no matter what, and i can’t think of a clean way to convey what this flag ought to be set to through create­Request­Action and the provided fetch function.

i think the only way to get rid of this flag is by also dropping it from wpcom-xhr-request, but i’m reluctant to do that [for node.js safety reasons](https://github.com/Automattic/wpcom-xhr-request/pull/31#pullrequestreview-543937363).

Reply:
i’ve since done exactly that in a68491234dee571a935ce536e10371480f010942 (thanks to @jsnajdr’s advice in <https://github.com/Automattic/wpcom-xhr-request/pull/31#discussion_r538245453>).